packages/block-library/src/post-template/edit.js

```diff
@@ -118,7 +118,7 @@ export default function PostTemplateEdit( {
 				postId: post.id,
 				queryId,
 			} ) ),
-		[ posts ]
+		[ posts, queryId ]
```

Comment:
Do we really need to pass the `queryId` in this context? I mean, Isn't it already present in the context?

Reply:
Here is the place we place we add all the context inside the `Post Template`, so we need to pass this as well.

Reply:
As far as I know `BlockContextProvider` is additive, meaning it concats the context with the previous context in the parent, and `queryId` comes from the parent context already. (Maybe I'm missing something though)

Reply:
I got it. Yes, it seems I don't have to pass this here, but we have to add it as a dependency.

Reply:
> I got it. Yes, it seems I don't have to pass this here, but we have to add it as a dependency.
 Write

That is weird, I don't know why we'd need this 🤔 since the value won't change.

Reply:
It seems really strange that we need to add queryId as a dependency, queryId is not used in the useMemo.
It may fix the bug because we will be generating a new object,  but it seems even without it things should work and we probably have an issue elsewhere.

Reply:
>Query Loop needs a unique id to handle multiple pagination properly and is using queryId attribute. This attribute is set on insertion/creation of the block after the first rendering.

The implementation [does this](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query/edit/index.js#L97), and that's why this problem doesn't occur always. I'm open to suggestions for alternatives though 😄

Reply:
ok, so personally when I try removing the queryId from the dependencies here, the fix is still valid.

Reply:
You're right. I started wrong on my previous error (adding it in this context). It's still a dependency in `BlockContextProvider`. Thanks!